include/RED4ext/Scripting/Natives/worldAnimationSystem.hpp

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <RED4ext/Common.hpp>
+#include <RED4ext/HashMap.hpp>
+#include <RED4ext/NativeTypes.hpp>
+#include <RED4ext/Scripting/Natives/AnimAnimatedObject.hpp>
+#include <RED4ext/Scripting/Natives/AnimMetaRig.hpp>
+#include <RED4ext/Scripting/Natives/Generated/Box.hpp>
+#include <RED4ext/Scripting/Natives/Generated/appearance/AppearanceResource.hpp>
+#include <RED4ext/Scripting/Natives/Generated/ent/RagdollComponent.hpp>
+#include <RED4ext/Scripting/Natives/Generated/world/IRuntimeSystem.hpp>
+#include <RED4ext/Scripting/Natives/animRig.hpp>
+#include <RED4ext/Scripting/Natives/entEntity.hpp>
+#include <RED4ext/SystemUpdate.hpp>
+#include <cstdint>
+
+namespace RED4ext
+{
+namespace ent
+{
+struct AnimatedComponent;
+struct EntityID;
+struct IPlacedComponent;
+struct ITransformAttachment;
+} // namespace ent
+namespace anim
+{
+struct AnimSetupResource;
+}
+
+namespace world
+{
+struct AnimationSystemScriptInterface;
+struct RuntimeSystemRendering;
+
+struct AnimatedEntitiesBucket
+{
+    struct ComponentBindings
+    {
+        struct Binding
+        {
+            Handle<ent::IPlacedComponent> placedComponent;         // 00
+            ent::AnimatedComponent* animComponent;                 // 10
+            Handle<ent::ITransformAttachment> transformAttachment; // 18
+        };
+        RED4EXT_ASSERT_SIZE(Binding, 0x28);
+
+        DynArray<Binding> bindings; // 00
+        uint64_t unk10;             // 10
+    };
+    RED4EXT_ASSERT_SIZE(ComponentBindings, 0x18);
+
+    static constexpr uint32_t MaxEntries = 2048;
+
+    HashMap<ent::EntityID, uint32_t> entityIDs;                          // 00
+    StaticArray<anim::AnimatedObject*, MaxEntries> animatedObjects;      // 30
+    StaticArray<Box, MaxEntries> animationBounds;                        // 4090
+    StaticArray<ent::AnimatedComponent*, MaxEntries> animatedComponents; // 140A0
+    StaticArray<Handle<ent::Entity>, MaxEntries> entities;               // 180A8
+    StaticArray<ComponentBindings, MaxEntries> componentBindings;        // 200B0
+    StaticArray<uint32_t, MaxEntries> unk2C0B8;                          // 2C0B8
+    uint64_t unk2E0C0[0x2E0E0 - 0x2E0C0];                                // 2E0C0
+};
+RED4EXT_ASSERT_SIZE(AnimatedEntitiesBucket, 0x2E170);
+
+struct RagdollEntitiesBucket
+{
+    StaticArray<anim::Rig*, 2048> rigs;                          // A2948
+    StaticArray<ent::RagdollComponent*, 2048> ragdollComponents; // A6950
+};
```

Comment:
Please add size assertion here.

Reply:
And if we use game's namespaces, let's add it to file names as for original classes too, i.e. `animKeyFrames.hpp`, `animAnimatedObject.hpp`, `animMetaRig.hpp`, `animAnimBufferState.hpp`, etc.